add OpenAISTTService

joachimchauvet opened the pull request:
This also works with OpenAI-compatible API providers such as Groq (requested in #254) by passing the `base_url`.

```python
stt = OpenAISTTService(
    api_key=os.getenv("GROQ_API_KEY"),
    base_url="https://api.groq.com/openai/v1",
    model="distil-whisper-large-v3-en"
)
```

aconchillo commented:
@joachimchauvet Thank you!!! Any chance we can rebase and try this again?

joachimchauvet commented:
@aconchillo I rebased and updated to use the new SegmentedSTTService ;)

aconchillo commented on examples/foundational/07g-interruptible-openai.py:
```diff
@@ -51,6 +50,8 @@ async def main():
             ),
         )
 
+        stt = OpenAISTTService(api_key=os.getenv("OPENAI_API_KEY"), model="whisper-1")
```
we should set `transcription_enabled=False` since we are using OpenAI's STT now.

aconchillo replied:
And we should enable audio passthrough `vad_audio_passthrough=True` so the audio can reach the STT.

aconchillo commented:
Thank you! Looks great! Added a couple of comments. Needs a rebase as well.

joachimchauvet commented:
I updated but some files currently already in the repo were preventing the checks from passing. I think the main culprit was from the recently merged OpenAI realtime PR.
Ruff also modified a bunch of other files. Let me know if you prefer me to exclude those @aconchillo

markbackman commented:
Superseded by: https://github.com/pipecat-ai/pipecat/pull/1183
